src/bp/core/services/telemetry/custom-roles.ts

```diff
@@ -0,0 +1,94 @@
+import { BUILTIN_MODULES } from 'common/defaults'
+import LicensingService from 'common/licensing-service'
+import { buildSchema } from 'common/telemetry'
+import { AuthRule, Workspace } from 'common/typings'
+import Database from 'core/database'
+import { calculateHash } from 'core/misc/utils'
+import { TelemetryRepository } from 'core/repositories/telemetry'
+import { TYPES } from 'core/types'
+import { inject, injectable } from 'inversify'
+import _ from 'lodash'
+import ms from 'ms'
+
+import { GhostService } from '..'
+import { JobService } from '../job-service'
+
+import { REDACTED, TelemetryStats } from './telemetry-stats'
+
+const DEFAULT_ROLES = ['dev', 'admin', 'editor']
+
+interface Role {
+  id: string
+  rules: AuthRule[]
+}
+
+interface WorkspacePayload {
+  id: string
+  roles: Role[]
+}
+
+const obsfuscateModule = rules => {
+  return rules.reduce((acc, rule) => {
+    const res = obfuscateRule(rule.res)
+    return [...acc, { ...rule, res }]
+  }, [])
+}
+
+const obfuscateRule = res => {
+  const regex = /^module\.([\w-]+)\.?/
+  const result = regex.exec(res)
+
+  if (!result) {
+    return res
+  }
+
+  return BUILTIN_MODULES.includes(result[1]) ? res : res.replace(result[1], REDACTED)
+}
+
+@injectable()
+export class RolesStats extends TelemetryStats {
+  protected interval: number
+  protected url: string
+  protected lock: string
+
+  constructor(
+    @inject(TYPES.GhostService) ghostService: GhostService,
+    @inject(TYPES.Database) database: Database,
+    @inject(TYPES.LicensingService) licenseService: LicensingService,
+    @inject(TYPES.JobService) jobService: JobService,
+    @inject(TYPES.TelemetryRepository) telemetryRepo: TelemetryRepository
+  ) {
+    super(ghostService, database, licenseService, jobService, telemetryRepo)
+    this.url = process.TELEMETRY_URL
```

Comment:
do stats classes ever use a custom or different url?

Reply:
yes, the legacy telemetry stats use a different url.